### PR TITLE
oauthclient: support for CERN authentication

### PR DESCRIPTION
**Pull request by jirikuncar**
* NEW Adds support for CERN authentication.  (PR #2821)

Co-authored-by: Pamfilos Fokianos <pamfilos.fokianos@cern.ch>
Signed-off-by: Jiri Kuncar <jiri.kuncar@cern.ch>

### CODE REVIEW

**Comment by jirikuncar** [comment]
@tiborsimko shall we include it in `2.0.1` or `2.1.0`?

**Comment by audub** [comment]
May I ask, why this is coded directly into invenio instead of into a CERN overlay? I thought the point was to keep site specific things in the overlay?

**Comment by jirikuncar** [comment]
@audub because we are planning to share the code among different services and it lives in `contrib` folder.

**Comment by PXke** [comment]
What does it mean that it lives in contrib ?

**Comment by jirikuncar** [comment]
> What does it mean that it lives in contrib?

`invenio/modules/oauthclient/`**`contrib`**`/cern.py`

**Comment by PXke** [comment]
Sure, but, what does it mean ?

**Comment by kaplun** [comment]
Googling around I found: http://blog.startifact.com/posts/against-contrib.html

which offers some food for thought :-)

**Comment by jirikuncar** [comment]
> which offers some food for thought :-)

Fine but this is not the place to discuss it moreover when we already have had discussion about this directory name before we put GitHub and ORCID in place (5c013b6). cc @lnielsen

**Comment by kaplun** [comment]
Eheh totally agree :smile: Was just surprised to find that thing as first result in google and was also an interesting reading to share.

**Comment by PXke** [comment]
Okay it makes sense so.

**Comment by jirikuncar** [comment]
@egabancho can you tested with real data? Btw can you post some **anonymized** responses from `https://oauthresource.web.cern.ch/api/Me`?

**Comment by egabancho** [comment]
We will set it up for CDSLabs this week and will post some data asap

File: invenio/modules/oauthclient/contrib/cern.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    return result
+
+
+def account_info(remote, resp):
+    """Retrieve remote account information used to find local user."""
+    # Query CERN Resources to get user info and groups
+    response = remote.get('https://oauthresource.web.cern.ch/api/Me')
+    res = get_dict_from_response(response)
+    email = res['http://schemas.xmlsoap.org/claims/EmailAddress']
+    nickname = res['http://schemas.xmlsoap.org/claims/Firstname']
```
Comment (jirikuncar) [inline invenio/modules/oauthclient/contrib/cern.py]: @ludmilamarian has fixed this ...

**Comment by jirikuncar** [comment]
I'm closing it in favor of @ludmilamarian's upcoming fixes.

**Comment by tiborsimko** [comment]
:shipit:

**Pull request merged**